raft_test.go

```diff
@@ -3270,3 +3270,45 @@ func TestRaft_runFollower_ReloadTimeoutConfigs(t *testing.T) {
 	// Check the follower loop set the right state
 	require.Equal(t, Candidate, env.raft.getState())
 }
+
+func TestRaft_PreVote_ShouldNotRejectLeader(t *testing.T) {
```

Comment:
maybe we can add a similar scenario in here that test the opposite: `TestRaft_PreVote_ShouldRejectNonLeader`